src/overlays/actors/ovl_En_Po_Sisters/z_en_po_sisters.c

```diff
@@ -192,673 +218,700 @@ void EnPoSisters_Destroy(Actor* thisx, PlayState* play) {
+ *  This is never reached because conditions are never met in EnPoSisters_AimlessIdleFlying
+ */
+void EnPoSisters_SetupAimlessIdleFlying2(EnPoSisters* this) {
+    Animation_MorphToLoop(&this->skelAnime, &gPoeSistersSwayAnim, -3.0f);
+    this->idleFlyingAnimationCounter = Rand_S16Offset(2, 3);
     this->actor.speedXZ = 0.0f;
-    this->actionFunc = func_80B1ABB8;
+    this->actionFunc = EnPoSisters_AimlessIdleFlying2;
 }
 
-void func_80B1ABB8(EnPoSisters* this, PlayState* play) {
+void EnPoSisters_AimlessIdleFlying2(EnPoSisters* this, PlayState* play) {
     SkelAnime_Update(&this->skelAnime);
     if (Animation_OnFrame(&this->skelAnime, 0.0f)) {
-        if (this->unk_192 != 0) {
-            this->unk_192--;
-        }
+        DECR(this->idleFlyingAnimationCounter);
     }
 
-    if ((this->unk_192 == 0) || (this->actor.xzDistToPlayer < 600.0f)) {
-        func_80B1AC40(this);
+    if ((this->idleFlyingAnimationCounter == 0) || (this->actor.xzDistToPlayer < 600.0f)) {
+        EnPoSisters_SetupAimlessIdleFlying(this);
     }
 }
 
-void func_80B1AC40(EnPoSisters* this) {
-    if (this->actionFunc != func_80B1AE3C) {
-        Animation_MorphToLoop(&this->skelAnime, &object_po_sisters_Anim_000D40, -3.0f);
+/**
+ *  Flying in a straight line, until wall or player comes near.
+ */
+void EnPoSisters_SetupAimlessIdleFlying(EnPoSisters* this) {
+    if (this->actionFunc != EnPoSisters_Investigating) {
+        Animation_MorphToLoop(&this->skelAnime, &gPoeSistersFloatAnim, -3.0f);
     }
-    this->unk_192 = Rand_S16Offset(15, 3);
-    this->unk_191 |= (0x6 | 0x1);
-    this->actionFunc = func_80B1ACB8;
+    this->idleFlyingAnimationCounter = Rand_S16Offset(15, 3);
+    this->poSisterFlags |= (POSISTERS_FLAG_CHECK_Z_TARGET | POSISTERS_FLAG_UPDATE_SHAPE_ROT | POSISTERS_FLAG_CHECK_AC);
+    this->actionFunc = EnPoSisters_AimlessIdleFlying;
 }
 
-void func_80B1ACB8(EnPoSisters* this, PlayState* play) {
+void EnPoSisters_AimlessIdleFlying(EnPoSisters* this, PlayState* play) {
     SkelAnime_Update(&this->skelAnime);
     Math_StepToF(&this->actor.speedXZ, 1.0f, 0.2f);
     if (Animation_OnFrame(&this->skelAnime, 0.0f)) {
-        if (this->unk_192 != 0) {
-            this->unk_192--;
-        }
+        DECR(this->idleFlyingAnimationCounter);
     }
 
     if ((this->actor.xzDistToPlayer < 600.0f) && (fabsf(this->actor.playerHeightRel + 5.0f) < 30.0f)) {
-        func_80B1AE28(this);
-    } else if ((this->unk_192 == 0) && Math_StepToF(&this->actor.speedXZ, 0.0f, 0.2f)) {
-        func_80B1AB5C(this);
+        EnPoSisters_SetupInvestigating(this);
+    } else if ((this->idleFlyingAnimationCounter == 0) && Math_StepToF(&this->actor.speedXZ, 0.0f, 0.2f)) {
+        // ! @bug: this is never reached because speedXZ is reduced
+        //         at the same rate it is increased at the top of this function
+        EnPoSisters_SetupAimlessIdleFlying2(this);
     }
 
-    if (this->actor.bgCheckFlags & 8) {
+    if (this->actor.bgCheckFlags & 8) { // wall
```

Comment:
```suggestion
    if (this->actor.bgCheckFlags & 8) { // touching a wall
```
This is the comment you made below, just for consistency